### PR TITLE
feat(accident-notification): Use hash do determine if attachment has been sent before

### PR DESCRIPTION
**Pull request by obmagnusson**
# ...

Attach a link to issue if relevant

## What

In attachment review the form was sending to the external api other documents on each submit. Use hashes by filecontent to determine if the file has been sent and does not send them.

## Why

Reviewers at SÍ where complaining that they have to go over the same document over and over again each time a user added a document

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/template-api-modules/src/lib/modules/templates/accident-notification/attachments/attachment.utils.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+export const attachmentStatusToAttachmentRequests = (
+  receivedAttachments?: AccidentNotificationAttachmentStatus,
+): AccidentNotificationAttachmentGatherRequest[] => {
+  if (!receivedAttachments) return allAttachmentRequestConfig.requests
+
+  const attachmentRequests: AccidentNotificationAttachmentGatherRequest[] = []
+
+  if (
+    !receivedAttachments.InjuryCertificate &&
+    receivedAttachments.InjuryCertificate != null
```
Comment (bensijons) [inline libs/application/template-api-modules/src/lib/modules/templates/accident-notification/attachments/attachment.utils.ts:29]: Couldn't you just do this? I feel like it is more readable
```suggestion
     receivedAttachments?.InjuryCertificate === false
```

File: libs/application/template-api-modules/src/lib/modules/templates/accident-notification/attachments/attachment.utils.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  ) {
+    attachmentRequests.push(injuryCertificateRequest)
+  }
+  if (
+    !receivedAttachments.ProxyDocument &&
+    receivedAttachments.ProxyDocument != null
```
Comment (bensijons) [inline libs/application/template-api-modules/src/lib/modules/templates/accident-notification/attachments/attachment.utils.ts:35]: same here

File: libs/application/template-api-modules/src/lib/modules/templates/accident-notification/attachments/attachment.utils.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  ) {
+    attachmentRequests.push(powerOfAttorneyRequest)
+  }
+  if (
+    !receivedAttachments.PoliceReport &&
+    receivedAttachments.PoliceReport != null
+  ) {
+    attachmentRequests.push(policeReportRequest)
+  }
+  if (!receivedAttachments.Unknown && receivedAttachments.Unknown != null) {
```
Comment (bensijons) [inline libs/application/template-api-modules/src/lib/modules/templates/accident-notification/attachments/attachment.utils.ts:45]: aand here

File: libs/application/template-api-modules/src/lib/modules/templates/accident-notification/attachments/attachment.utils.ts (hunk 1 above)
Comment (obmagnusson) [reply libs/application/template-api-modules/src/lib/modules/templates/accident-notification/attachments/attachment.utils.ts:29]: Well the receivedAttachement object is not null it is the InjuryCertificate that the null check is for.

Comment (obmagnusson) [reply libs/application/template-api-modules/src/lib/modules/templates/accident-notification/attachments/attachment.utils.ts:29]: Tests fail if i change the implementation to this suggestion

**Review by bensijons — APPROVED** [review]

**Pull request merged**